python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -192,6 +194,23 @@ def test_job_config_conda_env(conda_envs, shutdown_only):
         ray.shutdown()
 
 
+@pytest.mark.skipif(
+    os.environ.get("CONDA_DEFAULT_ENV") is None,
+    reason="must be run from within a conda environment")
+@pytest.mark.skipif(
+    os.environ.get("CI") and sys.platform != "linux",
+    reason="This test is only run on linux CI machines.")
+def test_job_config_conda_env_eagerly(conda_envs, shutdown_only):
+    runtime_env = {"conda": f"package-{REQUEST_VERSIONS[0]}"}
+    job_config = ray.job_config.JobConfig(
+        runtime_env=runtime_env, prepare_runtime_env_eagerly=True)
```

Comment:
I think we would like to move runtime_env out of the JobConfig and just have it specified in `ray.init()`.  `runtime_env` is already supported as a keyword argument in `ray.init()`.  

Is it possible to make this feature work with `ray.init(runtime_env=runtime_env, job_config=ray.job_config.JobConfig(prepare_runtime_env_eagerly=True)`?  Or even simpler, `ray.init(runtime_env=runtime_env, prepare_runtime_env_eagerly=True)`?  cc @edoakes, what are your thoughts on the API here?

Reply:
Yep, I'm also confused that there are two fields to set runtime_env. I agree that we should reserve only one. Let's make it clearly.